src/store/modules/order/actions.ts

```diff
@@ -10,8 +10,109 @@ import emitter from '@/event-bus'
 import store from "@/store";
 import { prepareOrderQuery } from "@/utils/solrHelper";
 import logger from "@/logger";
+import { getOrderCategory } from "@/utils/order";
 
 const actions: ActionTree<OrderState , RootState> ={
+
+  async getOrderDetails({commit}, payload ) {
+    // Show loader only when new query and not the infinite scroll
+    // if (payload.viewIndex === 0) 
+    emitter.emit("presentLoader");
+    let resp;
+    const orderQueryPayload = prepareOrderQuery({
+      ...payload,
+      orderStatusId: 'ORDER_APPROVED',
+      orderTypeId: 'SALES_ORDER',
+      '-fulfillmentStatus': '(Cancelled OR Rejected)',
+    })
+
+    try {
+      resp = await OrderService.getOpenOrders(orderQueryPayload);
+      if (resp.status === 200 && !hasError(resp) && resp.data.grouped?.orderId?.ngroups > 0) {
+        const orderIds = resp.data.grouped?.orderId?.groups.map((order: any) => order.doclist.docs[0].orderId);
+        const { productId, ...params } = payload;
+        this.dispatch('order/getOtherOrderItem', {...params, orderIds});
+      } else {  
+        showToast(translate("Orders Not Found"))
+      }
+    } catch(err) {
+      logger.error(err)
+      showToast(translate("Something went wrong"))
+    }
+    emitter.emit("dismissLoader");
+    return resp;
+  },
+
+  async getOtherOrderItem({ commit }, payload) {
+    // Show loader only when new query and not the infinite scroll
+    // if (payload.viewIndex === 0) 
+    emitter.emit("presentLoader");
+    let resp;
+    const orderQueryPayload = prepareOrderQuery({
+      ...payload,
+      orderStatusId: 'ORDER_APPROVED',
+      orderTypeId: 'SALES_ORDER',
+    })
+
+    try {
+      resp = await OrderService.getOpenOrders(orderQueryPayload);
+      if (resp.status === 200 && !hasError(resp) && resp.data.grouped?.orderId?.ngroups > 0) {
+        const orders = resp.data.grouped?.orderId?.groups.map((order: any) => {
+          const orderItem = order.doclist.docs[0]
+          return {
+            orderId: orderItem.orderId,
+            orderName: orderItem.orderName,
+            customer: {
+              partyId: orderItem.customerId,
+              name: orderItem.customerName
+            },
+            category: getOrderCategory(orderItem),
+            parts: order.doclist.docs.reduce((arr: Array<any>, item: any) => {
```

Comment:
Instead of creating parts as we don't need to display information on UI ion the basis of parts, create two separate objects `otherItems` and `currentItem`, where otherItems will have all the items other than the current displayed item and the currentItem will include the information for the current item.

otherItems = order.doclist.docs.filter((item: any) => item.productId != productId)
currentItem = order.doclist.docs.filter((item: any) => item.productId == productId)